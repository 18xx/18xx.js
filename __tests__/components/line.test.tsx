import * as React from 'react';
import { ReactElement } from 'react';
import * as renderer from 'react-test-renderer';

import Line from '../../src/components/line';

import Point from '../../src/point';

describe('Line', () => {
  describe('#toString', () => {
    it('returns an svg path element', () => {
      const subject: ReactElement<Line> = (
        <Line
          point1={new Point(1, 2)}
          point2={new Point(3, 4)}
        />
      );
      expect(renderer.create(subject)).toMatchSnapshot();
    });
  });
});
