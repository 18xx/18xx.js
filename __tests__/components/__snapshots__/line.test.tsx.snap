// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Line #toString returns an svg path element 1`] = `
<line
  stroke="black"
  strokeDasharray="1,0"
  strokeWidth={8}
  x1={1}
  x2={3}
  y1={2}
  y2={4}
/>
`;
